Please fix the lint errors 


Run npm run lint

> backend@0.0.1 lint
> eslint "{src,apps,libs,test}/**/*.ts" --fix


/home/runner/work/Lumenpulse/Lumenpulse/apps/backend/src/common/interceptors/idempotency.interceptor.spec.ts
Error:     6:10  error  'of' is defined but never used                                                                                       @typescript-eslint/no-unused-vars
Error:    10:7   error  'cacheService' is assigned a value but never used                                                                    @typescript-eslint/no-unused-vars
Error:    44:5   error  Use "@ts-expect-error" instead of "@ts-ignore", as "@ts-ignore" will do nothing if the following line is error-free  @typescript-eslint/ban-ts-comment
Error:    97:14  error  'e' is defined but never used                                                                                        @typescript-eslint/no-unused-vars
Error:   109:5   error  Unsafe return of a value of type `any`                                                                               @typescript-eslint/no-unsafe-return

/home/runner/work/Lumenpulse/Lumenpulse/apps/backend/src/common/interceptors/idempotency.interceptor.ts
Error:    39:11  error    Unsafe assignment of an `any` value                                               @typescript-eslint/no-unsafe-assignment
Warning:    47:27  warning  Unsafe argument of type `any` assigned to a parameter of type `string`            @typescript-eslint/no-unsafe-argument
Error:    47:35  error    Unsafe member access .method on an `any` value                                    @typescript-eslint/no-unsafe-member-access
Error:    52:11  error    Unsafe assignment of an `any` value                                               @typescript-eslint/no-unsafe-assignment
Error:    52:36  error    Unsafe member access .headers on an `any` value                                   @typescript-eslint/no-unsafe-member-access
Error:    59:49  error    Unsafe member access .body on an `any` value                                      @typescript-eslint/no-unsafe-member-access
Error:    60:45  error    Unsafe member access .path on an `any` value                                      @typescript-eslint/no-unsafe-member-access
Error:    85:13  error    Unsafe assignment of an `any` value                                               @typescript-eslint/no-unsafe-assignment
Error:    86:7   error    Unsafe call of an `any` typed value                                               @typescript-eslint/no-unsafe-call
Error:    86:16  error    Unsafe member access .status on an `any` value                                    @typescript-eslint/no-unsafe-member-access
Error:    95:9   error    Promise-returning function provided to property where a void return was expected  @typescript-eslint/no-misused-promises
Error:    96:17  error    Unsafe assignment of an `any` value                                               @typescript-eslint/no-unsafe-assignment
Error:    98:13  error    Unsafe assignment of an `any` value                                               @typescript-eslint/no-unsafe-assignment
Error:    98:34  error    Unsafe member access .statusCode on an `any` value                                @typescript-eslint/no-unsafe-member-access
Error:    99:13  error    Unsafe assignment of an `any` value                                               @typescript-eslint/no-unsafe-assignment
Error:   105:9   error    Promise-returning function provided to property where a void return was expected  @typescript-eslint/no-misused-promises

✖ 21 problems (20 errors, 1 warning)

Error: Process completed with exit code 1.

--- a/apps/backend/src/common/interceptors/idempotency.interceptor.spec.ts
+++ b/apps/backend/src/common/interceptors/idempotency.interceptor.spec.ts
@@ -3,11 +3,10 @@ import { IdempotencyInterceptor } from './idempotency.interceptor';
 import { Reflector } from '@nestjs/core';
 import { CacheService } from '../../cache/cache.service';
 import { ExecutionContext, CallHandler } from '@nestjs/common';
-import { of, throwError } from 'rxjs';
+import { throwError } from 'rxjs';
 
 describe('IdempotencyInterceptor', () => {
   let interceptor: IdempotencyInterceptor;
-  let cacheService: CacheService;
 
   const mockCacheService = {
     get: jest.fn(),
@@ -29,7 +28,6 @@ describe('IdempotencyInterceptor', () => {
     }).compile();
 
     interceptor = module.get<IdempotencyInterceptor>(IdempotencyInterceptor);
-    cacheService = module.get<CacheService>(CacheService);
   });
 
   it('should return cached result if key and body hash match', async () => {
@@ -37,13 +35,14 @@ describe('IdempotencyInterceptor', () => {
     const context = createMockContext('POST', { 'idempotency-key': 'test-key' }, body);
     const next: CallHandler = { handle: jest.fn() };
     
-    // @ts-ignore
+    // @ts-expect-error - accessing private method for test
     const bodyHash = interceptor.calculateHash(body);
     const cachedResponse = { statusCode: 201, body: { success: true }, bodyHash };
 
     mockCacheService.get.mockResolvedValue(cachedResponse);
 
-    const result = await (await interceptor.intercept(context, next)).toPromise();
+    const obs = await interceptor.intercept(context, next);
+    const result = await obs.toPromise();
     expect(result).toEqual(cachedResponse.body);
   });
 
@@ -70,14 +69,14 @@ describe('IdempotencyInterceptor', () => {
     try {
       const obs = await interceptor.intercept(context, next);
       await obs.toPromise();
-    } catch (e) {
+    } catch {
       // Expected
     }
 
     expect(mockCacheService.del).toHaveBeenCalled();
   });
 
-  function createMockContext(method: string, headers: any, body: any): ExecutionContext {
+  function createMockContext(method: string, headers: Record<string, string>, body: unknown): ExecutionContext {
     return {
       switchToHttp: () => ({
         getRequest: () => ({
@@ -93,6 +92,6 @@ describe('IdempotencyInterceptor', () => {
       }),
       getHandler: () => ({}),
       getClass: () => ({}),
-    } as any;
+    } as unknown as ExecutionContext;
   }
 });

--- a/apps/backend/src/common/interceptors/idempotency.interceptor.ts
+++ b/apps/backend/src/common/interceptors/idempotency.interceptor.ts
@@ -8,6 +8,7 @@ import {
   Logger,
 } from '@nestjs/common';
 import { Reflector } from '@nestjs/core';
+import { Request, Response } from 'express';
 import { Observable, of } from 'rxjs';
 import { tap } from 'rxjs/operators';
 import * as crypto from 'crypto';
@@ -19,7 +20,7 @@ import {
 
 interface IdempotencyResult {
   statusCode: number;
-  body: any;
+  body: unknown;
   bodyHash: string;
 }
 
@@ -35,12 +36,14 @@ export class IdempotencyInterceptor implements NestInterceptor {
   async intercept(
     context: ExecutionContext,
     next: CallHandler,
-  ): Promise<Observable<any>> {
-    const request = context.switchToHttp().getRequest();
-    const options = this.reflector.getAllAndOverride<IdempotentOptions>(
-      IDEMPOTENT_OPTIONS_KEY,
-      [context.getHandler(), context.getClass()],
-    ) || {};
+  ): Promise<Observable<unknown>> {
+    const httpContext = context.switchToHttp();
+    const request = httpContext.getRequest<Request>();
+    const options =
+      this.reflector.getAllAndOverride<IdempotentOptions>(
+        IDEMPOTENT_OPTIONS_KEY,
+        [context.getHandler(), context.getClass()],
+      ) || {};
 
     const methods = options.methods || ['POST', 'PUT', 'DELETE', 'PATCH'];
     if (!methods.includes(request.method)) {
@@ -50,14 +53,14 @@ export class IdempotencyInterceptor implements NestInterceptor {
     const headerName = options.header || 'idempotency-key';
     const idempotencyKey = request.headers[headerName.toLowerCase()];
 
-    if (!idempotencyKey) {
+    if (!idempotencyKey || typeof idempotencyKey !== 'string') {
       return next.handle();
     }
 
     // Hash the body to ensure the key is only valid for the same payload
     const bodyHash = this.calculateHash(request.body);
     const cacheKey = `idempotency:${request.path}:${idempotencyKey}`;
-    
+
     const cachedResult = await this.cacheService.get<IdempotencyResult | string>(
       cacheKey,
     );
@@ -71,7 +74,7 @@ export class IdempotencyInterceptor implements NestInterceptor {
       }
 
       const result = cachedResult as IdempotencyResult;
-      
+
       // Verify that the body hash matches
       if (result.bodyHash !== bodyHash) {
         throw new HttpException(
@@ -81,7 +84,7 @@ export class IdempotencyInterceptor implements NestInterceptor {
       }
 
       this.logger.debug(`Returning cached result for key: ${idempotencyKey}`);
-      const response = context.switchToHttp().getResponse();
+      const response = httpContext.getResponse<Response>();
       response.status(result.statusCode);
       return of(result.body);
     }
@@ -91,25 +94,33 @@ export class IdempotencyInterceptor implements NestInterceptor {
 
     return next.handle().pipe(
       tap({
-        next: async (body) => {
-          const response = context.switchToHttp().getResponse();
+        next: (body: unknown) => {
+          const response = httpContext.getResponse<Response>();
           const result: IdempotencyResult = {
             statusCode: response.statusCode,
             body,
             bodyHash,
           };
           const ttl = options.ttl || 24 * 60 * 60 * 1000; // 24 hours default
-          await this.cacheService.set(cacheKey, result, ttl);
+          this.cacheService
+            .set(cacheKey, result, ttl)
+            .catch((err: Error) =>
+              this.logger.error(`Failed to cache idempotency result: ${err.message}`),
+            );
         },
-        error: async () => {
+        error: () => {
           // Remove the lock on error so the client can retry
-          await this.cacheService.del(cacheKey);
-        }
-      })
+          this.cacheService
+            .del(cacheKey)
+            .catch((err: Error) =>
+              this.logger.error(`Failed to release idempotency lock: ${err.message}`),
+            );
+        },
+      }),
     );
   }
 
-  private calculateHash(body: any): string {
+  private calculateHash(body: unknown): string {
     const data = body ? JSON.stringify(body) : '';
     return crypto.createHash('sha256').update(data).digest('hex');
   }

